src/actions/auth/registrationAction.js

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import RegistrationConstants from './actionTypes';
+
+export function registerUserSuccess(newUser) {
+  return {
+    type: RegistrationConstants.REGISTER_SUCCESS,
+    payload: newUser,
+  };
+}
+
+export function registerUserFail(errorMessage) {
+  return {
+    type: RegistrationConstants.REGISTER_FAILURE,
+    payload: errorMessage,
+  };
+}
+
+export function registerRequest(newUser) {
+  return {
+    type: RegistrationConstants.REGISTER_REQUEST,
+    payload: newUser,
+  };
+}
+
+export function registerUser(newUser) {
+  // eslint-disable-next-line func-names
+  return function (dispatch) {
+    dispatch(registerRequest(newUser));
+    return axios.post('https://ah-backend-kronos-staging.herokuapp.com/api/users/', newUser, {
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((response) => {
+        dispatch(registerUserSuccess(response.data));
+        toast.success('Account created. Check your email to verify', {
+          containerId: 'A',
+        });
+        setTimeout(() => {
```

Comment:
I don't think navigating using setTimeout is a good idea, there are better ways you can do it using ReactRouter

Reply:
Thanks @Yiga-fred this has been resolved.